src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersReservedSlotsBehavior.cs

```diff
@@ -61,14 +67,18 @@ private Task ProcessVersionAsync(INetworkPeer peer, VersionPayload version)
                 // it's not yet included in the ConnectorPeers collection.
                 // freeSlots returns the number of available slots, considering that passed peer is taking one slot.
                 int freeSlots = connector.MaxOutboundConnections - connector.ConnectorPeers.Count - 1;
-                if (freeSlots >= 1)
+
+                // Get the number of PH-enabled peers we are already connected to.
+                int phEnabledPeersConnected = connector.ConnectorPeers.Count(p => this.DoesPeerSupportsPH(p.PeerVersion));
+
+                if (freeSlots >= (MinimumRequiredPeerSupportingPH - phEnabledPeersConnected))
```

Comment:
Would we have a case where `freeSlots` lands up as zero and we don't have the requisite number of PH peers connected?

I am also assuming in the case where `freeSlots` = 0 and `phEnabledPeersConnected` = 3 that we drop the connection elsewhere as we would be exceeding the connection limit. The log message is misleading in that case though.